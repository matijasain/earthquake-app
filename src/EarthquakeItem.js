import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class EarthquakeItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const {
      earthquakeName,
      earthquakeDate,
      earthquakeMagnitude,
      earthquakeAlert,
      tsunami,
      earthquakeUrl,
      children,
    } = this.props;

    return (
      <div className="item">
        <div className="image">{children}</div>

        <div className="content">
          <h3 className="header">{earthquakeName}</h3>
          <div className="meta">
            <span>Date: {moment(earthquakeDate).format('YYYY-MM-DD')}</span>
          </div>
          <div className="description">
            <p>Magnitude: {earthquakeMagnitude}</p>
            <p>Alert: {earthquakeAlert}</p>
            <p>Tsunami: {tsunami}</p>
          </div>
          <div className="extra">
            <a
              href={earthquakeUrl}
              alt="earthquake url"
              target="_blank"
              rel="noopener noreferrer"
            >
              {earthquakeUrl}
            </a>
          </div>
        </div>
      </div>
    );
  }
}

EarthquakeItem.propTypes = {
  earthquakeName: PropTypes.string.isRequired,
  earthquakeDate: PropTypes.number.isRequired,
  earthquakeMagnitude: PropTypes.number.isRequired,
  earthquakeAlert: PropTypes.string,
  tsunami: PropTypes.number.isRequired,
  earthquakeUrl: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

EarthquakeItem.defaultProps = {
  earthquakeAlert: null,
};

export default EarthquakeItem;
