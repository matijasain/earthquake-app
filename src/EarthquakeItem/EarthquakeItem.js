import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class EarthquakeItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const {
      earthquakeName,
      earthquakeDate,
      earthquakeMagnitude,
      earthquakeAlert,
      tsunami,
      earthquakeUrl,
      children,
    } = this.props;

    return (
      <div className="item">
        <div className="image">{children}</div>

        <div className="content">
          <h3 className="header">{earthquakeName}</h3>
          <div className="meta">
            <span>Date: {moment(earthquakeDate).format('YYYY-MM-DD')}</span>
          </div>
          <div className="description">
            <p>Magnitude: {earthquakeMagnitude}</p>
            <p>Alert: {earthquakeAlert}</p>
            <p>Tsunami: {tsunami}</p>
          </div>
          <div className="extra">
            <a
              href={earthquakeUrl}
              alt="earthquake url"
              target="_blank"
              rel="noopener noreferrer"
            >
              {earthquakeUrl}
            </a>
          </div>
        </div>
      </div>
    );
  }
}

EarthquakeItem.propTypes = {
  earthquakeName: PropTypes.string,
  earthquakeDate: PropTypes.number,
  earthquakeMagnitude: PropTypes.number,
  earthquakeAlert: PropTypes.string,
  tsunami: PropTypes.number,
  earthquakeUrl: PropTypes.string,
  children: PropTypes.node,
};

EarthquakeItem.defaultProps = {
  earthquakeName: '',
  earthquakeAlert: '',
  earthquakeDate: 1577833200000,
  earthquakeMagnitude: 0,
  tsunami: 0,
  earthquakeUrl: '',
  children: undefined,
};

export default EarthquakeItem;
