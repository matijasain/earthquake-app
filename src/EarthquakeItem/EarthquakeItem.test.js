import React from 'react';
import ReactDOM from 'react-dom';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

import EarthquakeItem from './EarthquakeItem';

afterEach(cleanup);

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EarthquakeItem />, div);
});

it('Matches snapshot', () => {
  const mainContainerSnap = renderer.create(<EarthquakeItem />).toJSON();
  expect(mainContainerSnap).toMatchSnapshot();
});
