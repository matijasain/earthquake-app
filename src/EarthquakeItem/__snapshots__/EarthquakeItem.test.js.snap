// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="item"
>
  <div
    className="image"
  />
  <div
    className="content"
  >
    <h3
      className="header"
    >
      
    </h3>
    <div
      className="meta"
    >
      <span>
        Date: 
        2020-01-01
      </span>
    </div>
    <div
      className="description"
    >
      <p>
        Magnitude: 
        0
      </p>
      <p>
        Alert: 
        
      </p>
      <p>
        Tsunami: 
        0
      </p>
    </div>
    <div
      className="extra"
    >
      <a
        alt="earthquake url"
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        
      </a>
    </div>
  </div>
</div>
`;
